libs/sdk-core/src/invoice.rs

```diff
@@ -189,6 +195,25 @@ pub fn add_lsp_routing_hints(
     Ok(invoice_builder.build_raw()?)
 }
 
+// Validate that the BOLT11 payment request matches the network
+pub fn validate_network(bolt11: &str, network: Network) -> InvoiceResult<()> {
+    if bolt11.trim().is_empty() {
```

Comment:
We duplicate the parse_invoice code here. Perhaps we should just use it and compare the network with the LNInvoice object?